### ISSUE TITLE
Setup Gradle Build Tool

### ISSUE THREAD

**Issue by nirrek** [romain77150/team-no-name#1]
Need a build tool for the project.

**Comment by romain77150** [comment]
Done, gradle is set up @nirrek can you please double check what i've done

**Comment by nirrek** [comment]
:+1: I'll pull it down an have a look this evening

**Comment by nirrek** [comment]
It was mostly correct, but with a few issues. I've actually cleaned things up a bit in a new repo which we should use instead. I've also added a readme with some setup instructions in this new repo.

The main changes I have made:

1. Made a proper .gitignore so that files and directories that shouldn't be in the repository aren't getting committed (eg you had committed eclipse config files, OSX Finder metadata files, etc.)

2. Fixed up the gradle dependencies so that the serial library from processing was a dependency (previously the file had been copied into the project directory)

3. You were declaring strange package names. Eg. if you had a class called Modeler you were putting it in a package main.java.modeler. The package in this case should just be defined as `package main`. In the new repo I'm actually just using the default package to keep things simple (I don't think there will be many files so it shouldn't require multiple packages or anything). Easy to change if need be.

Anyways new repo is here:
https://github.com/nirrek/bodyTracker

**Comment by romain77150** [comment]
yeah i browsed the processing.org website i couldn't find any proper library for the serial, just the file Serial.java. How did you find the repo for the Serial dependency? on www.mvnrepository.com?

I declared strange names because when i was reading the "beginners" tutorial on grade.org it said that, by default, gradle will search for the source files in src/main/java

Can i ask where you got those gradlew and gradlew.bat files from? I'm assuming you didn't code that all by yourself just to kill time :)
And what is the purpose of having a setting.gradle file?

**Comment by nirrek** [comment]
### Re: finding the serial dependency for Processing:
http://search.maven.org/#search%7Cga%7C1%7Cprocessing%20serial


### Re: directory structure
The default directory structure for a gradle project is what you read:

| Directory | Purpose |
--------------- | ---------------------------
|`src/main/java` | java source files |
| `src/test/java` | java unit tests |

The default structure actually is using the `default package` in Java so there doesn't need to be any package declaration in the source files.

My guess as to what is causing confusion for you is that you are thinking that the root directory for the source code is the project root. The actual root directory for the java source code is `src/main/java`. Therefore the `fully-qualified-package-name` of any class will be with respect to that directory. As a result you wouldn't put a package definition like `package src.main.java;` for .java files sitting inside `src/main/java`. They don't need any package namespacing as they are sitting in the root of the package directory hierarchy.

### Re: your doubts about the `main` directory. 
`src/main/java  // 1.`
`src/test/java   // 2.`

So why do we have a directory structure like this?
For 1. we are saying that our main source code that is written in Java is in this location.
For 2. we are saying that our test source code that is written in Java is in this location.
So by using `src` we are saying that it is source code. By using `/main` or `/test` we are saying whether its main program source code, or testing source code. And by using `/java` we are saying the source is java source code.

### Re: gradlew/gradlew.bat
See here http://gradle.org/docs/current/userguide/gradle_wrapper.html

### Re: settings.gradle
This is the settings file for Gradle. It mainly comes into play when doing multi-package builds. Its not strictly necessary in our current project structure, but the file will be generated by default when setting up a new gradle project. https://gradle.org/docs/current/userguide/build_lifecycle.html

**Comment by romain77150** [comment]
Thanks a lot for the clarifications mate!
I should definitely get some practice on my own, its not normal that in 3rd year im still confused with such basic stuffs... Then again they are not properly teaching us all that at uq

**Comment by nirrek** [comment]
no worries.

Also, you should delete this repository so that the others don't get confused between this one and the new one.
